apps/apps.go

```diff
@@ -0,0 +1,161 @@
+package apps
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	client "github.com/deis/controller-sdk-go"
+	"github.com/deis/controller-sdk-go/api"
+)
+
+const workflowURLPrefix = "deis."
+
+// List lists apps on a Deis controller.
+func List(c *client.Client, results int) ([]api.App, int, error) {
+	body, count, err := c.LimitedRequest("/v2/apps/", results)
+
+	if err != nil {
+		return []api.App{}, -1, err
+	}
+
+	var apps []api.App
+	if err = json.Unmarshal([]byte(body), &apps); err != nil {
+		return []api.App{}, -1, err
+	}
+
+	for name, app := range apps {
+		// Add in app URL based on controller hostname, port included
+		app.URL = fmt.Sprintf("%s.%s", app.ID, strings.TrimPrefix(c.ControllerURL.Host, workflowURLPrefix))
+		apps[name] = app
+	}
+
+	return apps, count, nil
+}
+
+// New creates a new app.
+func New(c *client.Client, id string) (api.App, error) {
+	body := []byte{}
+
+	var err error
+	if id != "" {
+		req := api.AppCreateRequest{ID: id}
+		body, err = json.Marshal(req)
+
+		if err != nil {
+			return api.App{}, err
+		}
+	}
+
+	resBody, err := c.BasicRequest("POST", "/v2/apps/", body)
+
+	if err != nil {
+		return api.App{}, err
+	}
+
+	app := api.App{}
+	if err = json.Unmarshal([]byte(resBody), &app); err != nil {
+		return api.App{}, err
+	}
+
+	// Add in app URL based on controller hostname, port included
+	app.URL = fmt.Sprintf("%s.%s", app.ID, strings.TrimPrefix(c.ControllerURL.Host, workflowURLPrefix))
+
+	return app, nil
+}
+
+// Get app details from a Deis controller.
+func Get(c *client.Client, appID string) (api.App, error) {
+	u := fmt.Sprintf("/v2/apps/%s/", appID)
+
+	body, err := c.BasicRequest("GET", u, nil)
+
+	if err != nil {
+		return api.App{}, err
+	}
+
+	app := api.App{}
+
+	if err = json.Unmarshal([]byte(body), &app); err != nil {
+		return api.App{}, err
+	}
+
+	// Add in app URL based on controller hostname, port included
+	app.URL = fmt.Sprintf("%s.%s", app.ID, strings.TrimPrefix(c.ControllerURL.Host, workflowURLPrefix))
+
+	return app, nil
+}
+
+// Logs retrieves logs from an app.
+func Logs(c *client.Client, appID string, lines int) (string, error) {
+	u := fmt.Sprintf("/v2/apps/%s/logs", appID)
+
+	if lines > 0 {
+		u += "?log_lines=" + strconv.Itoa(lines)
+	}
+
+	body, err := c.BasicRequest("GET", u, nil)
+
+	if err != nil || len(body) < 1 {
+
+		return "", errors.New(
```

Comment:
@Joshua-Anderson can you pull this error out into a global, exported `var`? since it's an SDK, it would be helpful if callers can check the error value and decide, programmatically, if there are no log messages